docs/source/features.rst

```diff
@@ -3,15 +3,44 @@ Features
 
 Dynamic Source Tree Detection
 =============================
-coming soon
+Fab analyses your source code to determine the Fortran compile order,
+which means you don't have to manually specify and maintain the compile order,
+which can become problematic in a large project.
+
 
 Incremental Build
 =================
-coming soon
+Fab "watermarks" each artefact with a checksum of its inputs.
+Subsequent builds avoid reprocessing by searching for watermarks in the prebuild folder.
+
+
+Sharing Prebuilds
+=================
+Artefacts from previous builds can be shared between users,
+either by copying the prebuild folder or using the :class:`~fab.steps.grab.GrabPreBuild` step.
+
+
+Metrics
+=======
+Fab will record the time taken by each step and plot them on a pie chart.
+
+.. image:: img/pie.png
+    :width: 66%
+    :alt: pie chart
```

Comment:
The example plot shows the utility of the feature. Is there way to get separate metrics for both passes of two pass compilation?

Reply:
The histogram and busby plots are produced for both stages. These are "sub-step" metrics, the top-level pie chart only currently shows the total for each whole step. We could always refactor the metrics to do it though.